### PR TITLE
Fix #2682: Make getBaseDomain use LRU cache

### LINKED ISSUE #2682: Make getBaseDomain use LRU cache
@diracdeltas mentions this function is slow and we call it during filtering so we should be using an LRU cache for the lookups.

### PR DESCRIPTION
**Pull request by willy-b**
- [x ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.

Should the refererExceptions array also be updated to include the base domains that keep popping up? (E.g. facebook.com, naver.net, etc.)

### CODE REVIEW

File: js/lib/baseDomain.js [hunk 1]
```diff
@@ -44,5 +54,8 @@ module.exports.getBaseDomain = function (hostname) {
     tld--
   }
 
-  return curDomain
+  baseDomain = curDomain
```
Comment (bbondy) [inline js/lib/baseDomain.js]: not a big deal and super nit but you can probably just remove this line and use curDomain for the below 2 lines.

File: app/filtering.js [hunk 2]
```diff
@@ -196,7 +196,7 @@ function registerForBeforeSendHeaders (session) {
           requestHeaders['Cookie'] = undefined
         }
         if (requestHeaders['Referer'] &&
-            !refererExceptions.includes(parsedUrl.hostname)) {
+            !refererExceptions.includes(parsedUrl.hostname) && !refererExceptions.includes(getBaseDomain(parsedUrl.hostname))) {
```
Comment (bbondy) [inline app/filtering.js]: cc @diracdeltas This task is good on its own without this. We don't need both checks here.  Would you prefer we update `refererExceptions` to the base domains and use `getBaseDomain` here or would you prefer we use the more narrow exceptions with not modifying `refererExceptions` and not using `getBaseDomain`?

**Comment by bbondy** [comment]
Thanks for doing this! Just have a question for Yan first and then this will need a slight modification one way or the other and then we can land.

File: app/filtering.js (hunk 2 above)
Comment (diracdeltas) [inline app/filtering.js]: i think the way it was before is fine

**Comment by willy-b** [comment]
Sweet. Thanks for running a project that makes it so easy to contribute! 

Once I know whether to use refererExceptions or getBaseDomain I'll make that change and address the nit you pointed out earlier.

**Comment by willy-b** [comment]
filtering.js is reverted per @diracdeltas  and a redundant line is avoided per @bbondy

**Comment by bbondy** [comment]
++ thanks!

**Comment by luixxiul** [comment]
In 0.11.2?

**Pull request merged**